app-libs/exchange-oracle/src/coin_market_cap.rs

```diff
@@ -0,0 +1,219 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::sgx_reexport_prelude::*;
+
+use crate::{error::Error, exchange_rate_oracle::OracleSource, types::TradingPair, ExchangeRate};
+use itc_rest_client::{http_client::HttpClient, rest_client::RestClient, RestGet, RestPath};
+use lazy_static::lazy_static;
+use log::*;
+use serde::{Deserialize, Serialize};
+use std::{
+	collections::{BTreeMap, HashMap},
+	env,
+	string::{String, ToString},
+	time::Duration,
+};
+use url::Url;
+
+const COINMARKETCAP_URL: &str = "https://pro-api.coinmarketcap.com";
+const COINMARKETCAP_KEY_PARAM: &str = "CMC_PRO_API_KEY";
+const FIAT_CURRENCY_PARAM: &str = "convert_id";
+const CRYPTO_CURRENCY_PARAM: &str = "id";
+const COINMARKETCAP_PATH: &str = "v2/cryptocurrency/quotes/latest"; //For free plan
```

Comment:
`//For free plan`

What does this comment mean?

Reply:
Is `COINMARKETCAP_PATH` itself not descriptive enough?

Reply:
The focus is on defining the path. There are other API endpoints to get the exchange rate, which however require a paid plan

Reply:
Ok, then I agree on `// API endpoint to get the exchange rate with a basic API plan (free)`